### PR TITLE
Replace outcome::result by ErrorMessageOr in OrbitGgp

### PR DESCRIPTION
**Pull request by beckerhe**
Bug: http://b/196022630

### CODE REVIEW

File: src/OrbitGgp/InstanceTests.cpp [hunk 1]
```diff
@@ -65,8 +66,8 @@ TEST(InstanceTests, GetListFromJson) {
                           "key\":\"object value\"}}]")
                           .toUtf8();
     const auto result = Instance::GetListFromJson(json);
-    ASSERT_FALSE(result);
-    EXPECT_EQ(result.error().value(), static_cast<int>(orbit_ggp::Error::kUnableToParseJson));
+    ASSERT_TRUE(result.has_error());
+    EXPECT_THAT(result.error().message(), testing::HasSubstr("Unable to parse JSON"));
```
Comment (dimitry-) [inline src/OrbitGgp/InstanceTests.cpp:70]: EXPECT_THAT(result, HasError());

Comment (beckerhe) [reply src/OrbitGgp/InstanceTests.cpp]: Ah yes, Thanks. Fixed.

File: src/OrbitGgp/InstanceTests.cpp [hunk 2]
```diff
@@ -23,21 +25,21 @@ TEST(InstanceTests, GetListFromJson) {
...
-    ASSERT_TRUE(empty_instances);
+    ASSERT_TRUE(empty_instances.has_value());
     EXPECT_TRUE(empty_instances.value().empty());
   }
 
   {
     // one empty json object
     const auto json = QString("[{}]").toUtf8();
-    EXPECT_FALSE(Instance::GetListFromJson(json));
+    EXPECT_TRUE(Instance::GetListFromJson(json).has_error());
```
Comment (dimitry-) [inline src/OrbitGgp/InstanceTests.cpp:42]: EXPECT_THAT(..., HasError("..."));

File: src/OrbitGgp/InstanceTests.cpp [hunk 3]
```diff
@@ -93,8 +94,8 @@ TEST(InstanceTests, GetListFromJson) {
                           "key\":\"other value\",\"other complex object\":{\"object "
                           "key\":\"object value\"}}]")
                           .toUtf8();
-    const auto result = Instance::GetListFromJson(json);
-    ASSERT_TRUE(result);
+    auto result = Instance::GetListFromJson(json);
+    ASSERT_TRUE(result.has_value());
```
Comment (dimitry-) [inline src/OrbitGgp/InstanceTests.cpp:98]: ASSERT_THAT(result, HasNoError()); // this has added benefit of automatically printing the error message in the case assert fais.

File: src/OrbitGgp/SshInfoTests.cpp [hunk 4]
```diff
@@ -15,23 +15,29 @@
...
 
 TEST(SshInfoTest, CreateFromJson) {
-  QByteArray json;
-
   // Empty json
-  json = QString("").toUtf8();
-  EXPECT_FALSE(SshInfo::CreateFromJson(json));
+  {
+    QByteArray json = QString("").toUtf8();
+    EXPECT_TRUE(SshInfo::CreateFromJson(json).has_error());
```
Comment (dimitry-) [inline src/OrbitGgp/SshInfoTests.cpp:21]: same here and below, HasError("...")

File: src/OrbitGgp/SshInfoTests.cpp [hunk 5]
```diff
@@ -43,32 +49,36 @@ TEST(SshInfoTest, CreateFromJson) {
...
-  EXPECT_EQ(ssh_info.port, 11123);
-  EXPECT_EQ(ssh_info.user, "a username");
+  {
+    QByteArray json = QString(
+                          "{\"host\":\"1.1.0.1\",\"keyPath\":\"/usr/local/some/path/.ssh/"
+                          "id_rsa\",\"knownHostsPath\":\"/usr/local/another/path/"
+                          "known_hosts\",\"port\":\"11123\",\"user\":\"a username\"}")
+                          .toUtf8();
+    const auto ssh_info_result = SshInfo::CreateFromJson(json);
+    ASSERT_TRUE(ssh_info_result.has_value());
```
Comment (dimitry-) [inline src/OrbitGgp/SshInfoTests.cpp:59]: HasNoError()

File: src/OrbitGgp/SshInfoTests.cpp [hunk 6]
```diff
@@ -43,32 +49,36 @@ TEST(SshInfoTest, CreateFromJson) {
...
-  // This is supposed to fail, since its expected that the port is a string
-  EXPECT_FALSE(SshInfo::CreateFromJson(json));
+  {
+    QByteArray json = QString(
+                          "{\"host\":\"1.1.0.1\",\"keyPath\":\"/usr/local/some/path/.ssh/"
+                          "id_rsa\",\"knownHostsPath\":\"/usr/local/another/path/"
+                          "known_hosts\",\"port\":11123,\"user\":\"a username\"}")
+                          .toUtf8();
+    // This is supposed to fail, since its expected that the port is a string
+    EXPECT_TRUE(SshInfo::CreateFromJson(json).has_error());
```
Comment (dimitry-) [inline src/OrbitGgp/SshInfoTests.cpp:80]: HasError(...)

File: src/OrbitGgp/InstanceTests.cpp (hunk 2 above)
Comment (beckerhe) [reply src/OrbitGgp/InstanceTests.cpp]: done.

File: src/OrbitGgp/InstanceTests.cpp (hunk 3 above)
Comment (beckerhe) [reply src/OrbitGgp/InstanceTests.cpp]: done.

File: src/OrbitGgp/SshInfoTests.cpp (hunk 4 above)
Comment (beckerhe) [reply src/OrbitGgp/SshInfoTests.cpp]: done.

File: src/OrbitGgp/SshInfoTests.cpp (hunk 5 above)
Comment (beckerhe) [reply src/OrbitGgp/SshInfoTests.cpp]: done.

File: src/OrbitGgp/SshInfoTests.cpp (hunk 6 above)
Comment (beckerhe) [reply src/OrbitGgp/SshInfoTests.cpp]: done.

**Review by dimitry- — APPROVED** [review]

**Comment by beckerhe** [comment]
Thanks a lot Dimitry for taking the time.

**Pull request merged**